### PR TITLE
Refactor DAG run metadata and add it to merges

### LINKED ISSUE #57: Automatically add DAG metadata to MergeOperator
Same as #47, but for merges.

### PR DESCRIPTION
**Pull request by arielshaqed**
Fixes #57.

### CODE REVIEW

File: lakefs_provider/operators/with_metadata_operator.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        Dict has keys 'endpoint_url', 'id', 'run_id', 'logical_date'.
+        """
+        ret = {}
+        ret['endpoint_url'] = airflow_conf.get("webserver", "base_url")
+
+        # TODO(ariels): Configurably filter metadata keys.
+        for key in self.__required_keys:
+            value = context.get(key, None)
+            if value is None:
+                self.log.warning(f"context does not contain {key}")
```
Comment (Jonathan-Rosenberg) [inline lakefs_provider/operators/with_metadata_operator.py:45]: if they are required, isn't it an error if they're not found?

File: lakefs_provider/operators/with_metadata_operator.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            ret[key] = value
+
+        return ret
+
+    def enrich_metadata(self, metadata: Dict[str, Any], context: Dict[str, Any]):
```
Comment (Jonathan-Rosenberg) [inline lakefs_provider/operators/with_metadata_operator.py:50]: don't think you use the provided `metadata` dict

**Review by Jonathan-Rosenberg — CHANGES_REQUESTED** [review]
Thank you!
A comment and a question and we're done :-)

File: lakefs_provider/operators/with_metadata_operator.py (hunk 1 above)
Comment (arielshaqed) [reply lakefs_provider/operators/with_metadata_operator.py:45]: No, it just means that we cannot generate the metadata.  We can still commit or merge; I don't want to fail that.

File: lakefs_provider/operators/with_metadata_operator.py (hunk 2 above)
Comment (arielshaqed) [reply lakefs_provider/operators/with_metadata_operator.py]: Ouch, true.  `self.metadata` is weird and always confuses me.

Removed, thanks!

**Review by arielshaqed — COMMENTED** [review]
Thanks!

PTAL...

**Review by Jonathan-Rosenberg — APPROVED** [review]
Thanks!

**Comment by arielshaqed** [comment]
Thanks!

**Pull request merged**